### PR TITLE
[Plot] - Remove PlotMetadata

### PR DESCRIPTION
**Pull request by bluong**
This field just isn't really something that should be exposed to the user.

### CODE REVIEW

File: src/core/interfaces.ts [hunk 1]
```diff
@@ -3,13 +3,13 @@ module Plottable {
...
   export interface Accessor<T> {
-    (datum: any, index: number, dataset: Dataset, plotMetadata: Plots.PlotMetadata): T;
+    (datum: any, index: number, dataset: Dataset): T;
   }
 
   /**
    * Retrieves scaled datum property.
    */
-  export type _Projector = (datum: any, index: number, dataset: Dataset, plotMetadata: Plots.PlotMetadata) => any;
+  export type _Projector = (datum: any, index: number, dataset: Dataset) => any;
```
Comment (jtlan) [inline src/core/interfaces.ts]: Do we still need the `_Projector` type, by the way?

**Comment by jtlan** [comment]
[+1] Minor comment about `_Projector` type, but that probably belongs in a different PR anyway.

File: src/core/interfaces.ts (hunk 1 above)
Comment (bluong) [inline src/core/interfaces.ts]: Now this is where type aliasing is probably handy.

`Accessor`s currently have the understanding that the scale has not affected their values whereas `_Projector`s do have that understanding that the scale has possibly affected the end value.

That said, if you think that differentiation is not enough of a merit for separation, possibly they can just be folded into `Accessor`

Comment (jtlan) [inline src/core/interfaces.ts]: Maybe `type`-name it something like `ScaledAccessor`, but only in the module where it's used?

**Pull request merged**